### PR TITLE
Provide detailed usage statistics

### PR DESCRIPTION
**Pull request by mrsimpson**
In order to determine the real usage and user attraction of Rocket.Chat, I as an operator need to know how users really interact with the system.
This PR adds information about how users interacted actively (wrote messages) or passively (read messages).

### CODE REVIEW

**Comment by mrsimpson** [comment]
This potentially creates high volumes, so a setting to turn this on/off might be useful. First wanted to get your (@RocketChat/core) feedback

**Comment by mrsimpson** [comment]
@geekgonecrazy yup. But once an hour is not really often for a DB. And the aggregate shall always hit indices. I really think that’s what a DB should be able to manage easily. Would you mind executing the query on open manually once and let us know the figures?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by geekgonecrazy** [comment]
This is true looks like mostly indexed.  @sampaiodiego what do you think?

**Comment by mrsimpson** [comment]
@geekgonecrazy also note that the „biggest“ cardinals to is subscriptions. Imho, this should be manageable by any DB

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sampaiodiego** [comment]
wow, that's scarey on a first look.. need more time look carefully 😄

File: packages/rocketchat-statistics/server/functions/getUsages.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+			user.reads = {};
+		}
+		if (writes.has(user._id)) {
+			user.writes = writes.get(user._id);
+			active = true;
+		} else {
+			user.writes = {};
+		}
+		user.active = active;
+		user._id = SHA256(user.emails.address);
```
Comment (mrsimpson) [inline packages/rocketchat-statistics/server/functions/getUsages.js]: This is only for obfuscation: Exporting who has done what on a system is quite some sensitive information. Hashing the email-address will allow a stable reference while at the same time not allowing the admin / receiver of the statistic to control a person's work.

**Comment by mrsimpson** [comment]
@sampaiodiego scary because of the DB-layer-implementation?
We thought it was more appropriate to have the DB handle aggregations and joins compared tohaving this done on the appserver. This way, the DB engine can optimize and parallelize.
Actually, it's quite simple, but pushing code down always comes with the penalty of complexity

**Comment by geekgonecrazy** [comment]
In the DB is for sure more efficient then in code.  No doubt at all.

I think just scary because new DB queries that go over a bulk of data is a bit scary.  😁 More places to overload the database. 

I think might be worth it, especially if off by default and we could start to see how it performs

**Comment by mrsimpson** [comment]
Closing since we faced performance issue on very large installations.
An external, async solution seems to be favorable (as foreseen by you guys already)

**Pull request closed without merging**